ApiSurface/DocCoverage.fs

```diff
@@ -99,10 +99,12 @@ module DocCoverage =
         let types =
             assembly.GetTypes ()
             |> Array.filter (fun ty ->
-                // Skip the new code-analysis types added in net7 (discussion ongoing: see
-                // https://github.com/dotnet/runtime/issues/93699 )
+                // A version of the F# compiler in 7.0.400 emitted some generated types which were not marked
+                // as compiler-generated. Skip those.
+                // https://github.com/dotnet/fsharp/issues/16141
                 not (ty.FullName.StartsWith ("System.Diagnostics.CodeAnalysis.Dynamic", StringComparison.Ordinal))
             )
+            |> Array.filter (fun ty -> obj.ReferenceEquals (ty.GetCustomAttribute<CompilerGeneratedAttribute> (), null))
```

Comment:
Why put these in two separate `Array.filter`s? I appreciate first is temporary, but I would prefer not to construct a new array for the intermediary filter

Reply:
Sure - though the first is not actually temporary, we'll have to keep it as long as 7.0.4xx is supported.

I just thought it was more aesthetically pleasing really. I'll change it when I get back home.